### PR TITLE
AF: vol-12x + eval-every-3 + WD=5e-3 (lighter regularization)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Combine vol-12x (programme-best surface weight) + eval-every-3 (throughput trick) + WD=5e-3 (lighter regularization from rei #3238 which achieved surface=0.000249). The triple combination addresses all three promising signals simultaneously.

## Instructions
```bash
cd target/icml2026 && python train.py --dataset airfrans --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 5e-3 \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 --enable-fourier \
  --epochs 999 --ema-decay 0.999 --use-lookahead --compile-model \
  --vol-loss-weight 12 --eval-interval 3 --save-checkpoint \
  --wandb-group af-eval-every-3-vol-sweep
```
**Target:** surface_mse < 0.000296 AND vol_mse < 0.002039

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT edward:

## Results

### Summary
**Did not beat baseline.** The 4L/256d Lion + vol-12x + eval-every-3 + WD=5e-3 combination converges well initially but diverges catastrophically after ~196 epochs due to escalating gradient instability at cosine LR peaks.

### Best Metrics (epoch 180)
| Metric | This Run | Baseline (PR #3135) | Delta |
|--------|----------|-------------------|-------|
| val_primary/surface_mse | **0.001365** | 0.000296 | +4.6x (worse) |
| full_val/volume_mse | **0.006940** (ep174) | 0.002039 | +3.4x (worse) |

### Training Trajectory
- **Epochs 1-138:** Smooth convergence with expected cosine oscillations. New best every ~3-6 eval cycles.
- **Epochs 138-180:** Increasing oscillation amplitude at cosine LR peaks. Surface best improved from 0.00148 (ep138) to 0.00136 (ep180), but with growing instability between cycles.
- **Epochs 180-196:** Gradient explosions at LR peaks becoming severe (grad_norm spikes: 97 → 180 → 265 → 366).
- **Epoch 196+:** Full divergence. Loss regressed to epoch ~40 levels (1.72). EMA model unable to buffer sustained instability.

### Configuration
```bash
cd target/icml2026 && python train.py --dataset airfrans --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 5e-3 \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 --enable-fourier \
  --epochs 999 --ema-decay 0.999 --use-lookahead --compile-model \
  --volume-loss-weight 12 --eval-interval 3 --save-checkpoint \
  --wandb-group af-eval-every-3-vol-sweep \
  --agent edward --wandb-name "edward/af-vol12x-eval3-wd5e3"
```

- **W&B run:** 00pefyfs (`edward/af-vol12x-eval3-wd5e3`)
- **Peak memory:** ~40.8 GB (GPU 0, RTX PRO 6000 Blackwell 98GB)
- **Training time:** ~3 hours at divergence, still running

### What Happened
The triple combination (vol-12x + eval-every-3 + WD=5e-3) with Lion optimizer and **4 layers** initially showed promising convergence but ultimately failed to reach baseline-competitive metrics. Three key issues:

1. **4L depth + cosine LR peaks = instability.** The 4-layer model is significantly more sensitive to LR peaks than the baseline 2L model. Gradient norms at cosine peaks escalated each cycle (40 → 60 → 100 → 180 → 265 → 366), eventually causing divergence.

2. **Lion + high LR for 4L is too aggressive.** The baseline uses AdamW at lr=6e-4 with 2L. This run uses Lion at lr=1.25e-4 with 4L. Despite the lower LR, the deeper model amplifies gradient instability. Lion's sign-based updates may exacerbate this.

3. **WD=5e-3 insufficient regularization for 4L.** The lighter weight decay (vs baseline WD=1e-2) may have reduced the stability margin for the deeper model.

**Note:** The PR command used `--vol-loss-weight` which doesn't exist — corrected to `--volume-loss-weight`.

### Suggested Follow-ups
- **Try the same triple combination with 2L/256d** (matching baseline depth). The 4L depth appears to be the instability source, not the other hyperparameters.
- **Lower the LR for 4L** (e.g., lr=5e-5) or **use gradient clipping < 1.0** (gc=0.5 was stable for 4L in earlier AirfRANS experiments per BASELINE.md).
- **Reduce T_max for 4L** (e.g., T_max=10 or T_max=25). Shorter cosine cycles would limit the duration of destabilizing LR peaks.
- **Try WD=1e-2 with 4L** to see if the baseline regularization strength prevents the divergence.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3345 because ADVISOR: Closing — vol-12x + eval-3 + WD=5e-3 diverged at ep196 (gradient explosion at cosine restart). Surface 4.6x worse, vol 3.4x worse. Note: command used --volume-loss-weight — verify this is the correct flag name (should be --vol-loss-weight). WD=5e-3 + high vol-weight appears unstable. Edward reassigned to DM metric-aware sweep..
